File: BE/src/main/java/com/haivn/common_api/SinhVienDangKy.java
package com.haivn.common_api;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.DynamicUpdate;
import org.hibernate.annotations.Where;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Table(name = "sinh_vien_dang_ky")
@Getter
@Setter
@DynamicUpdate
@Where(clause = "deleted=false")
public class SinhVienDangKy extends BaseEntity{
    @Column(name = "full_name")
    private String fullName;
    @Column(name = "ma_sv")
    private String maSV;
    @Column(name = "id_lop")
    private Long idLop;
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="id_lop",referencedColumnName="id", nullable = false, insertable = false, updatable = false)
    private LopHoc lopHoc;
    @Column(name = "ngay_sinh")
    private Timestamp ngaySinh;
    @Column(name = "dia_chi")
    private String diaChi;
    @Column(name = "gioi_tinh")
    private Boolean gioiTinh;
    @Column(name = "email")
    private String email;
    @Column(name = "sdt")
    private String sdt;
    @Column(name = "mo_ta")
    private String moTa;
    @Column(name = "status")
    private Short status;
}

File: FE/lib/model/model_qltvn/sinh-vien-dang-ky.dart
// ignore_for_file: file_names

import 'package:haivn/model/model_qltvn/status.dart';

import 'lop_hoc.dart';

class SinhVienDangKy {
  int? id;
  String? fullName;
  String? maSV;
  int? idLop;
  LopHoc? lopHoc;
  String? ngaySinh;
  String? diaChi;
  bool? gioiTinh;
  String? email;
  String? sdt;
  int? status;
  Status? statusObj;

  SinhVienDangKy({this.id, this.fullName, this.maSV, this.idLop, this.lopHoc, this.ngaySinh, this.diaChi, this.gioiTinh, this.email, this.sdt, this.status, this.statusObj});

  SinhVienDangKy.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        fullName = json['fullName'],
        maSV = json['maSV'],
        idLop = json['idLop'],
        lopHoc = LopHoc.fromJson(json['lopHoc']),
        ngaySinh = json['ngaySinh'],
        diaChi = json['diaChi'],
        gioiTinh = json['gioiTinh'],
        email = json['email'],
        sdt = json['sdt'],
        status = json['status'],
        statusObj = Status(
            id: json['status'],
            name: (json['status'] == 0)
                ? "Đăng ký"
                : (json['status'] == 1)
                    ? "Chờ phỏng vấn"
                    : (json['status'] == 2)
                        ? "Đỗ"
                        : (json['status'] == 3)
                            ? "Trượt"
                            : "");

  Map<String, dynamic> toJson() => {'fullName': fullName, 'maSV': maSV, 'idLop': idLop, 'ngaySinh': ngaySinh, 'diaChi': diaChi, 'gioiTinh': gioiTinh, 'email': email, 'sdt': sdt, 'status': status};
}
